### PR TITLE
Add leader election

### PR DESCRIPTION
**Pull request by jnpacker**
Signed off by @jnpacker 

* Add support for leader election
* Increase replica's to two
* Update self build process

### CODE REVIEW

File: deploy/controller/template-deployment.yaml [hunk 1]
```diff
@@ -8,29 +8,40 @@ objects:
   kind: Deployment
   metadata:
     labels:
-      name: curator-controller
-    name: curator-controller
+      name: ccluster-curator-controller
```
Comment (hanqiuzh) [inline deploy/controller/template-deployment.yaml:11]: is this a typo?

File: deploy/controller/deployment.yaml [hunk 2]
```diff
@@ -19,12 +19,15 @@ spec:
         - ./cluster-curator-controller
         - "-v"
         - "2"
+        image: registry.ci.openshift.org/open-cluster-management/cluster-curator-controller:latest        - name: POD_NAME
```
Comment (hanqiuzh) [inline deploy/controller/deployment.yaml:22]: wrong line?

Comment (jnpacker) [reply deploy/controller/deployment.yaml:22]: yes. Good catch.

File: deploy/controller/deployment.yaml [hunk 3]
```diff
@@ -19,12 +19,15 @@ spec:
...
         env:
-        #- name: IMAGE_TAG
-        #  value: sha256:4e7ab8e1beefd3127bd3fe1313b16e105a8216e4cf76b19bdfde6f87a2a31b34
-        image: registry.ci.openshift.org/open-cluster-management/cluster-curator-controller:latest
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        image: quay.io/jpacker/cluster-curator-controller@sha256:90419fd30a086ab0519112bd764e6c772575df88435580b188960cc206b38442
```
Comment (jnpacker) [inline deploy/controller/deployment.yaml:28]: Actually this is wrong as well. There are two image definitions.

File: deploy/controller/deployment.yaml (hunk 2 above)
Comment (jnpacker) [reply deploy/controller/deployment.yaml]: ## Validation
rh:[🔱leader-election]:🏡/workspace_git/cluster-curator-controller$ oc apply -k deploy/controller/
serviceaccount/cluster-curator unchanged
clusterrole.rbac.authorization.k8s.io/cluster-curator configured
clusterrolebinding.rbac.authorization.k8s.io/cluster-curator unchanged
deployment.apps/cluster-curator-controller configured
rh:[🔱leader-election]:🏡/workspace_git/cluster-curator-controller$ oc delete -k deploy/controller/
serviceaccount "cluster-curator" deleted
clusterrole.rbac.authorization.k8s.io "cluster-curator" deleted
clusterrolebinding.rbac.authorization.k8s.io "cluster-curator" deleted
deployment.apps "cluster-curator-controller" deleted

File: deploy/controller/template-deployment.yaml (hunk 1 above)
Comment (jnpacker) [reply deploy/controller/template-deployment.yaml]: yes

**Review by hanqiuzh — APPROVED** [review]

**Pull request merged**